### PR TITLE
Fix unbound variables

### LINKED ISSUE #606: First level attribute can be unbound
There is a change that you never enter the while loop, which makes the `first_level` attribute undefined.

https://github.com/mamba-org/quetz/blob/efcfdb8259a44b6765f2962a29047889bc70f48e/quetz/config.py#L395-L408

This can cause the following error.

```
UnboundLocalError: cannot access local variable 'first_level' where it is not associated with a value
```

I feel like this should have been caught by some pre-commit hook (e.g. `mypy`).

### PR DESCRIPTION
**Pull request by janjagusch**
Depends on #607 
Closes #606 

This is a first attempt to fix all possibly unbound variables in the code. I've decided to use the most straightforward fix I could think of. But I feel like some of the code has structural problems and that a refactoring might be in order.

### CODE REVIEW

File: plugins/quetz_repodata_zchunk/tests/test_main.py [hunk 1]
```diff
@@ -142,7 +142,7 @@ def test_repodata_zchunk(
             with open('repodata.json') as f:
                 repodata_unzck = f.read()
 
-            assert repodata == repodata_unzck  # NOQA
+            assert repodata == repodata_unzck  # NOQA # type: ignore
```
Comment (janjagusch) [inline plugins/quetz_repodata_zchunk/tests/test_main.py:145]: This seems utterly broken. The `repodata` variable isn't defined anywhere. Do we run the plugin tests in CI?

File: quetz/cli.py [hunk 2]
```diff
@@ -546,6 +546,7 @@ def start(
             )
             raise typer.Abort()
 
+    supervisor_process = None
```
Comment (janjagusch) [inline quetz/cli.py:549]: I feel like there should be a more elegant pattern to do the same thing.

File: quetz/dao.py [hunk 3]
```diff
@@ -860,7 +861,9 @@ def create_version(
                     )
                     if is_newer:
                         break
-                version_order = v.version_order if is_newer else v.version_order + 1
+                version_order = (
+                    v.version_order if is_newer else v.version_order + 1  # type: ignore
+                )
```
Comment (janjagusch) [inline quetz/dao.py:866]: This needs to be improved.

File: quetz/utils.py [hunk 4]
```diff
@@ -160,6 +162,8 @@ def apply_custom_query(search_type, db, keywords, filters):
                 each_keyword_condition = collate(Channel.name, "und-x-icu").notlike(
                     f'%{negation_argument}%'
                 )
+            else:
+                raise KeyError(search_type)
```
Comment (janjagusch) [inline quetz/utils.py:166]: We should probably use dictionaries here.

File: plugins/quetz_repodata_zchunk/tests/test_main.py (hunk 1 above)
Comment (simonbohnen) [reply plugins/quetz_repodata_zchunk/tests/test_main.py:145]: We do, i can check this tomorrow 👍🏼

Comment (simonbohnen) [reply plugins/quetz_repodata_zchunk/tests/test_main.py:145]: Probably the `if` branch is never run, and the `else` branch doesn't contain any asserts 🤷🏼‍♂️

**Pull request merged**